src/Auth/AuthController.js

```diff
@@ -229,68 +151,117 @@ class AuthController {
       returnURL = locationHref + returnURL.substring(4);
     }
     return utils.cleanURLFromPrYvParams(returnURL);
-  }
 
-  /**
-   * Util to grab parameters from url query string
-   * @param {*} url 
-   */
-  static getServiceInfoFromURL (url) {
-    const queryParams = utils.getQueryParamsFromURL(url || window.location.href);
-    //TODO check validity of status
-    return queryParams[AuthController.options.SERVICE_INFO_QUERY_PARAM_KEY];
+    function returnUrlIsAuto (returnURL) {
+      return returnURL.indexOf(RETURN_URL_AUTO) === 0;
+    }
   }
 
-  /**
-   * Eventually return pollUrl when returning from login in another page
-   */
-  async pollUrlReturningFromLogin (url) {
-    const params = utils.getQueryParamsFromURL(url);
-    let pollUrl = null;
-    if (params.prYvkey) { // deprecated method - To be removed
-      pollUrl = this.pryvServiceInfo.access + params.prYvkey;
+  async startAuthRequest () {
+    this.state = await postAccess.call(this);
+    
+    await doPolling.call(this);
+    
+    async function postAccess () {
+      try {
+        const res = await utils.superagent
+          .post(this.serviceInfo.access)
+          .send(this.settings.authRequest);
+        return res.body;
+      } catch (e) {
+        this.state = {
+          status: AuthStates.ERROR,
+          message: 'Requesting access',
+          error: e
+        };
+        throw e; // forward error
+      }
     }
-    if (params.prYvpoll) {
-      pollUrl = params.prYvpoll;
+
+    async function doPolling() {
+      if (this.state.status !== AuthStates.NEED_SIGNIN) {
+        return;
+      }
+      const pollResponse = await pollAccess.call(this);
```

Comment:
this could be a bit simplified with await pollAccess(this.state.poll);

Reply:
yes